services/scan/src/cvrs/export.test.ts

```diff
@@ -0,0 +1,590 @@
+  mockOf(isFeatureFlagEnabled).mockImplementation(() => true);
+
+  const buildCastVoteRecordMock = jest.spyOn(
+    buildCastVoteRecord,
+    'buildCastVoteRecord'
+  );
+
+  const addBallotImagesToCvrMock = jest.spyOn(
+    buildCastVoteRecord,
+    'addBallotImagesToCvr'
+  );
+
+  const cvr: CastVoteRecord = {
+    _ballotStyleId: '12',
+    _ballotType: 'standard',
+    _batchId: 'cd79e271-d5a3-4755-9af4-dfb491b8bcf1',
+    _batchLabel: 'batch label',
+    _precinctId: '23',
+    _scannerId: '000',
+    _testBallot: false,
+    _pageNumbers: [1, 2],
+    'county-commissioners': ['witherspoonsmithson'], // an existing candidate, not a write-in
+  };
+
+  buildCastVoteRecordMock.mockReturnValue(cvr);
+
+  const frontNormalizedFilePath = tmp.tmpNameSync();
+  await copyFile(BlankJpegPath, frontNormalizedFilePath);
+
+  const backNormalizedFilePath = tmp.tmpNameSync();
+  await copyFile(BlankJpegPath, backNormalizedFilePath);
+
+  const store = Store.memoryStore();
+  store.setElection(stateOfHamilton.electionDefinition);
+
+  // No CVRs, export should be empty
+  let stream = new streams.WritableStream();
+  await pipeline(exportCastVoteRecordsAsNdJson({ store }), stream);
+  expect(stream.toString()).toEqual('');
+
+  const metadata: BallotPageMetadata = {
+    locales: { primary: 'en-US' },
+    electionHash: stateOfHamilton.electionDefinition.electionHash,
+    ballotType: BallotType.Standard,
+    ballotStyleId: stateOfHamilton.election.ballotStyles[0].id,
+    precinctId: stateOfHamilton.election.precincts[0].id,
+    isTestMode: false,
+    pageNumber: 1,
+  };
+
+  store.addHmpbTemplate(Buffer.of(1, 2, 3), metadata, [
+    {
+      pageSize: { width: 1, height: 1 },
+      metadata: {
+        ...metadata,
+        pageNumber: 1,
+      },
+      contests: [],
+    },
+    {
+      pageSize: { width: 1, height: 1 },
+      metadata: {
+        ...metadata,
+        pageNumber: 2,
+      },
+      contests: [],
+    },
+  ]);
+
+  // Create CVRs, confirm that they are exported should work
+  const batchId = store.addBatch();
+  const sheetId = store.addSheet(uuid(), batchId, [
+    {
+      originalFilename: '/tmp/front-page.png',
+      normalizedFilename: frontNormalizedFilePath,
+      interpretation: {
+        type: 'UninterpretedHmpbPage',
+        metadata: {
+          ...metadata,
+          pageNumber: 2,
+        },
+      },
+    },
+    {
+      originalFilename: '/tmp/back-page.png',
+      normalizedFilename: backNormalizedFilePath,
+      interpretation: {
+        type: 'UninterpretedHmpbPage',
+        metadata: {
+          ...metadata,
+          pageNumber: 1,
+        },
+      },
+    },
+  ]);
+  store.adjudicateSheet(sheetId);
+
+  stream = new streams.WritableStream();
+  await pipeline(exportCastVoteRecordsAsNdJson({ store }), stream);
+  expect(stream.toString()).toEqual(
+    expect.stringContaining(stateOfHamilton.election.precincts[0].id)
+  );
```

Comment:
I thought you had concerns about being able to test a `Readable` CVR export, but this seems pretty simple?

Reply:
Previously I was trying to test the internals of the stream because I was trying to build a `Readable` by hand. Now I'm just using the included utilities like `pipeline` because I'm trusting that `Readable.from` is doing the right thing.